samples/sample_encode/src/sample_encode.cpp

```diff
@@ -1433,7 +1433,7 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
     //    return MFX_ERR_UNSUPPORTED;
     //}
 
-    if ((pParams->nMaxSliceSize) && (pParams->CodecId != MFX_CODEC_AVC))
+    if ((pParams->nMaxSliceSize) && !((pParams->CodecId != MFX_CODEC_AVC) || (pParams->CodecId != MFX_CODEC_HEVC)))
```

Comment:
It is always false condition.

Following De Morgan's laws !(A || B) is same as !A && !B, i.e.
```
(pParams->CodecId == MFX_CODEC_AVC) && (pParams->CodecId == MFX_CODEC_HEVC)
```

I believe correct condition is
```
if ((pParams->nMaxSliceSize) && ((pParams->CodecId != MFX_CODEC_AVC) || (pParams->CodecId != MFX_CODEC_HEVC)))
```

Reply:
fixed